plugins/capture.rb

```diff
@@ -0,0 +1,605 @@
+      # See if there was a config file set
+      @@start_opt_parser.parse(args) do |opt, idx, val|
+        case opt
+        when '--configfile'
+          config_file = val
+        end
+      end
+
+      options = default_options
+      config_options = read_config(config_file)
+      options = options.merge(config_options)
+
+      @@start_opt_parser.parse(args) do |opt, idx, val|
+        case opt
+        when '--session'
+          options[:session] = val
+        when '-i'
+          options[:srvhost] = val
+        when '--spoofip'
+          options[:spoof_ip] = val
+        when '--regex'
+          options[:spoof_regex] = val
+        when '-v'
+          options[:verbose] = true
+        when '--basic'
+          options[:http_basic] = true
+        when '--cert'
+          options[:ssl_cert] = val
+        when '--stdout'
+          options[:stdout] = true
+        when '--logfile'
+          options[:logfile] = val
+        when '--hashdir'
+          options[:hashdir] = val
+        when '-h'
+          options[:show_help] = true
+        end
+      end
+
+      options
+    end
+
+    def poison_included(options)
+      poisoners = ['mDNS','LLMNR','NBNS']
+      poisoners.each do |svc|
+        if options[:services][svc]
+          return true
+        end
+      end
+      false
+    end
+
+    # Fill in implied parameters to make the running code neater
+    def transform_params(options)
+      # If we've been given a specific IP to listen on, use that as our poisoning IP
+      if options[:spoof_ip].nil? && options[:srvhost] != '0.0.0.0'
+        options[:spoof_ip] = options[:srvhost]
+      end
+      
+      unless options[:session].nil?
+        # UDP is not supported on remote sessions
+        udp = ['NBNS','LLMNR','mDNS','SIP']
+        udp.each do |svc|
+          if options[:services][svc]
+            print_line("Skipping #{svc}: UDP server not supported over a remote session")
+            options[:services][svc] = false
+          end
+        end
+      end
+
+      if options[:logfile].nil?
+        options[:logfile] = default_logfile(options)
+      end
+
+      if options[:hashdir].nil?
+        options[:hashdir] = default_hashdir(options)
+      end
+    end
+
+    def validate_params(options)
+      if options[:srvhost].nil?
+        raise ArgumentError.new('Must provide an IP address to listen on')
+      end
+      # If we're running poisoning (which is disabled remotely, so excluding that situation), 
+      # we need either a specific srvhost to use, or a specific spoof IP
+      if options[:spoof_ip].nil? && poison_included(options)
+        raise ArgumentError.new('Must provide a specific IP address to use for poisoning')
+      end
+      unless options[:ssl_cert].nil? || File.file?(options[:ssl_cert])
+        raise ArgumentError.new("File #{options[:ssl_cert]} not found")
+      end
+      unless options[:session].nil? || framework.sessions.key?(options[:session].to_i)
```

Comment:
```suggestion
      unless options[:session].nil? || framework.sessions.get(options[:session])
```

Using [`#get`](https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/session_manager.rb#L264) will allow the user to specify `-1` as a session to reference the last created session. It'll also convert it to an integer for you.